proxy/proxy.go

```diff
@@ -1125,6 +1114,9 @@ func (p *Proxy) do(ctx *context) error {
 		backendStart := time.Now()
 		rsp, perr := p.makeBackendRequest(ctx, backendContext)
 		if perr != nil {
+			if perr.code != 499 { // 499 is logged later with more context
+				p.log.Errorf("Failed to do backend request: %v", perr)
+			}
```

Comment:
I think the proxy error is logged in `errorResponse` in all cases.
Why do we need this special case? Maybe we could fix the failing test somehow instead?

Reply:
Yes true but if we do not log it here we won't see retries in the logs, so if we only see this message and not the other it was a successful retry.
On the other hand we also try to retry on 499 if I understand our current code correctly, which might be also something to fix :)

Reply:
fixed